### PR TITLE
Refactor of top level tests

### PR DESCRIPTION
**Pull request by whytheplatypus**
Example of running a local cross cloud test:
```
podman-compose -f docker-compose.yml -f docker-compose.azurite.yml -f docker-compose.minio.yml -f docker-compose.testing.yml up -d
```

> [!note]
A directory called test-bucket must be in the _data directory for the local volume for minio, where this is will differ between systems.

### CODE REVIEW

File: upload-server/cmd/main_test.go [hunk 1]
```diff
@@ -30,14 +33,47 @@ func TestTus(t *testing.T) {
...
+	os.Setenv("SERVER_PORT", fmt.Sprintf("%d", port))
+	os.Setenv("UPLOAD_CONFIG_PATH", "../../upload-configs")
+	os.Setenv("S3_ENDPOINT", "http://minio:8000")
+	os.Setenv("S3_BUCKET_NAME", "test-bucket")
+	os.Setenv("AWS_REGION", "us-east-1")
+	os.Setenv("AWS_ACCESS_KEY_ID", "minioadmin")
+	os.Setenv("AWS_SECRET_ACCESS_KEY", "minioadmin")
+
+	os.Setenv("EDAV_STORAGE_ACCOUNT", "devstoreaccount1")
+	os.Setenv("EDAV_STORAGE_KEY", "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==")
```
Comment (whytheplatypus) [inline upload-server/cmd/main_test.go:66]: this is the example key from azurite, [public here](https://github.com/Azure/Azurite?tab=readme-ov-file#default-storage-account)

**Review by thetif — APPROVED** [review]
looks good

**Review by cfarmer-fearless — CHANGES_REQUESTED** [review]
Minor cleanup but looks great overall

**Comment by cfarmer-fearless** [comment]
Getting this when running locally
`{"time":"2024-09-26T17:32:09.308398739Z","level":"ERROR","source":{"function":"github.com/cdcgov/data-exchange-upload/upload-server/cmd.main","file":"/code/cmd/main.go","line":92},"msg":"error starting app, error initialize dex handler","app_info":{"System":"
DEX","Product":"UPLOAD API","App":"UPLOAD SERVER","Env":"DEV"},"error":"failed to connect to edav deliverer target not enough information given to connect to container edav"}`

File: upload-server/docker-compose.testing.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+version: "3.9"
+services:
+  upload-server:
+    command: go test -coverprofile=c.out -coverpkg=./... ./... -tags=integration
+    environment:
+      - AZURITE_STORAGE_KEY=${AZURE_STORAGE_KEY}
```
Comment (cfarmer-fearless) [inline upload-server/docker-compose.testing.yml:6]: Where is this getting populated?

Comment (cfarmer-fearless) [reply upload-server/docker-compose.testing.yml:6]: Coming from command line

**Review by cfarmer-fearless — APPROVED** [review]

**Pull request merged**